src/SFA.DAS.EmployerIncentives.Api/Authorization/AuthorizationExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace SFA.DAS.EmployerIncentives.Api.Authorization
+{
+    public static class ApiAuthorizationExtensions
+    {
+        public static IServiceCollection AddApiAuthorization(this IServiceCollection services, IWebHostEnvironment environment)
+        {
+            var isDevelopment = environment.IsDevelopment();
+
+            services.AddAuthorization(o =>
+            {
+                o.AddPolicy("APIM", policy =>
+                {
+                    if (isDevelopment)
+                    {
+                        policy.AllowAnonymousUser();
+                    }
+                    else
+                    {
+                        policy.RequireAuthenticatedUser();
+                        policy.AddAuthenticationSchemes(JwtBearerDefaults.AuthenticationScheme);
+                        policy.RequireRole("APIM");
+                    }
+                });
+            });
+            if (isDevelopment)
+                services.AddSingleton<IAuthorizationHandler, LocalAuthorizationHandler>();
```

Comment:
Why is this required? Why not just not have Auth if you are local?

Reply:
This allow the Authentication pipe line to be called for an anonymous user in the same way as a authenticated user. In theory it will allow us to add the [Authorized] to controllers and it would still work. However, can we discuss the best way forward